File: DSA/Arshdeep-Kaur/Day 25/Day25_Question1.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int start=0, end=nums.size()-1;
        
        while (start <= end) {
            while (start <= end && nums[start] != val) {
                start++;
            }
            while (start <= end && nums[end] == val) {
                end--;
            }
            if (start < end) {
                swap(nums[start++], nums[end--]);
            }
        }

        return start;
    }
};

File: DSA/Manav-Rajput/Day25_question_1.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int count = 0;
        for (int i : nums) {
            if (i != val) {
                count++;
            }
        }
        int[] nums_new = new int[count];
        int index = 0;
        for (int i : nums) {
            if (i != val) {
                nums_new[index++] = i;
            }
        }
        for (int i = 0; i < nums_new.length; i++) {
            nums[i] = nums_new[i];
        }

        return count;
    }
}
